scripts/py_matter_yamltests/matter_yamltests/parser.py

```diff
@@ -303,17 +305,38 @@ def _update_with_definition(self, container: dict, mapping_type):
 
         for value in list(container['values']):
             for key, item_value in list(value.items()):
-                mapping = mapping_type if self.is_attribute else mapping_type[value['name']]
+                if self.is_attribute:
+                    mapping = mapping_type
+                else:
+                    target_key = value['name']
+                    if mapping_type.get(target_key) is None:
+                        for candidate_key in mapping_type:
+                            if candidate_key.lower() == target_key.lower():
```

Comment:
I am okay with this going in as is, but I was wondering do we want to potentially create a `get_case_insensitive`? Do we want to strongly enforce this value to be the same as what is defined in the XML?

Reply:
It does enforce consistency and I would prefer if we can do that. People tends to copy-paste a mistake over and over. If it really becomes too annoying and a pain for daily development I am fine to reconsider this. But in theory and per my understanding, the spec defines the name of the attribute (with the casing).